### ISSUE TITLE
[package] freetype/2.11.1: does not provide Freetype::Freetype for drop-in replacement

### ISSUE THREAD

**Issue by paulharris** [conan-io/conan-center-index#10687]
Using conan, but not using requires="freetype/2.11.1", so we use CMake's standard freetype finder.

---

The standard CMake's find_package(Freetype) will provide TARGET Freetype::Freetype
with INCLUDE_DIR

Using standard CMake's find_package(freetype) will provide TARGET freetype,
but will not provide any INCLUDE_DIR stuff.

---

Conan's find_package(Freetype) will provide TARGET freetype and NOT TARGET Freetype::Freetype
but freetype has INCLUDE_DIR

Using conan and find_package(freetype) will provide TARGET freetype
AND the INCLUDE_DIR stuff.

---

What we want is a drop-in replacement, which means it should behave the same as CMake standard.
meaning, find_package(Freetype) also provides INCLUDE_DIRS

```
I tried to fix it by adding:
self.cpp_info.set_property("cmake_target_aliases", ["Freetype::Freetype"])
```
And it works, but there is a warning about the alias clashing with an existing target name.
I'm not sure if we should add this alias, or to add INCLUDE_DIR to the existing Freetype::Freetype target.

### Package and Environment Details (include every applicable attribute)
  * Package Name/Version: **freetype/2.11.1**
  * Operating System+version: **Linux Ubuntu 20.04**
  * Compiler+version: **GCC 11**
  * Docker image: **-**
  * Conan version: **conan 1.48.0**
  * Python version: **Python 3.8.10**

### Conan profile (output of `conan profile show default` or `conan profile show <profile>` if custom profile is in use)
```
[settings]
compiler=gcc
os=Linux
os_build=Linux
arch=x86_64
arch_build=x86_64
compiler.version=11
compiler.libcxx=libstdc++11
compiler.cppstd=17
build_type=Release
[options]
[conf]
tools.cmake.cmaketoolchain:generator=Ninja
[build_requires]
[env]
```

### Steps to reproduce (Include if Applicable)
```
Make a simple recipe with "conan new testfreetype/1.0@ --template=cmake_exe

In conanfile.py, add
generators = "CMakeDeps"
requires = ["freetype/2.11.1"]

in CMakeLists.txt add:
find_package(Freetype)
#find_package(freetype)
target_link_libraries(testfreetype PUBLIC Freetype::Freetype)
#target_link_libraries(testfreetype PUBLIC freetype)



In the src/testfreetype.cpp add:
#include <ft2build.h>

In the system /usr/include/freetype2/ft2build.h add at the top:
#error "should not use system include"
```

```
TEST 1 (fails with conan dep)
Do a conan install + build
It will fail with:
No target found   "Freetype::Freetype"
```
```
TEST 2 (success with system dep)
In conanfile.py comment out:
# requires = ["freetype/2.11.1"]
Do a conan install + build
It will succeed, but with an error indicating we are using the system include (thanks to our #error)
```

Then continue to comment out various lines to see the combination of behaviours.

**Comment by SpaceIm** [comment]
As already explained on Slack (https://cpplang.slack.com/archives/C41CWV9HA/p1651818019947719), `freetype` recipe properly models module & config file:
- module: `FindFreetype.cmake`, target `Freetype::Freetype` (see https://cmake.org/cmake/help/latest/module/FindFreetype.html)
- config: `freetype-config.cmake`, target `freetype` (see https://github.com/freetype/freetype/blob/e8ebfe988b5f57bfb9a3ecb13c70d9791bce9ecf/CMakeLists.txt#L624-L632).

Your issue comes from CMakeToolchain forcing `CMAKE_FIND_PACKAGE_PREFER_CONFIG` to ON: https://github.com/conan-io/conan/issues/10387.

**Comment by paulharris** [comment]
I appreciate all your comments SpaceIm, very professional and helpful.
I thought it good to bring this onto the Issues list as it isn't so simple as disabling that flag,
and there seemed to be several things happening at once so it was getting very confusing.
I needed a place to write my thoughts down and share, and slack was only working halfway.

I didn't want to pollute the other issue with too much noise while I figured things out.

---

RE my comment, "it isn't so simple as disabling that flag"

For my example, if you do a conan install, and then comment out that flag,
it does find Freetype::Freetype, but it then fails to find Brotli (looking for Findbrotli.cmake)
```
CMake Error at /build/msrap2/devenv/share/cmake-3.23/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):                                                                                                    
No "Findbrotli.cmake" found in CMAKE_MODULE_PATH.
Call Stack (most recent call first):
build/FindFreetype.cmake:25 (find_dependency)
CMakeLists.txt:5 (find_package)

FindFreetype.cmake:25 is:
       find_dependency(${_DEPENDENCY} REQUIRED ${${_DEPENDENCY}_FIND_MODE}) 
which ends up calling
find_package(brotli REQUIRED;MODULE)
```

So I can see multiple problems that might need to be addressed, I've put my best guess solutions here as well because I don't like to drop problems on people without thoughts on a solution:

1) VTK injects its own Finders in the CMAKE_MODULE_PATH before conan's, so it was "working, sometimes" because it wasn't using CMakeDeps's finders.
Possible solution: patch VTK so it *appends* their cmake path, rather than prepend.  Might require a flag to choose that behaviour, just for conan?  I'd rather patch upstream where possible, particularly in this situation as VTK is the birthplace of CMake.

2) Broli must also provide a Module export, and not just a CONFIG export.   I see in the conan-io/conan#10387, there is a list of upgraded libraries but I guess Brotli still requires the upgrade.
Possible solution: upgrade Brotli CCI recipe.

3) Should CMakeDeps demand that MODULEs only require other MODULEs?
line tools/cmake/cmakedeps/templates/config.py:66
enforces consistent find_package() usage (CONFIG or MODULES).
I get the impression this is an attempt to massage Modern CMake practices though the ecosystem?   Or is this a mistake?

**Comment by SpaceIm** [comment]
> RE my comment, "it isn't so simple as disabling that flag"
> 
> For my example, if you do a conan install, and then comment out that flag, it does find Freetype::Freetype, but it then fails to find Brotli (looking for Findbrotli.cmake)
> 
> ```
> CMake Error at /build/msrap2/devenv/share/cmake-3.23/Modules/CMakeFindDependencyMacro.cmake:47 (find_package):                                                                                                    
> No "Findbrotli.cmake" found in CMAKE_MODULE_PATH.
> Call Stack (most recent call first):
> build/FindFreetype.cmake:25 (find_dependency)
> CMakeLists.txt:5 (find_package)
> 
> FindFreetype.cmake:25 is:
>        find_dependency(${_DEPENDENCY} REQUIRED ${${_DEPENDENCY}_FIND_MODE}) 
> which ends up calling
> find_package(brotli REQUIRED;MODULE)
> ```

Seems to be a bug in `CMakeDeps`, and it should be reported to conan client.

**Comment by paulharris** [comment]
Actually sorry, I checked again, this issue is *not* solved by conan-io/conan#11201

**Comment by paulharris** [comment]
I'm closing this, the bug is still there but I'll discuss on conan#11201
